tests/unit/test_read_configs_include.py

```diff
@@ -442,11 +463,10 @@ def test_not_allowed_substitution_all(self, read_config_include_file, exit_on_er
             command_suffix,
             exit_on_error=exit_on_error,
         )
-        expected = set()
-        for k in self.new_vars:
-            expected.add(f"{k}=new")
         actual = [re.sub(r"[\s\n]+", " ", val.strip()) for val in get_command_stdout(proc).split(ENV_SPLIT_STR_ALT)]
-        assert all(val != "new" for val in actual)
+        # "val" is like "NEW_VAR=" without the "new" value because it is initially unset and
+        # old var are not allowed to override so it stays unset.
+        assert all("new" not in val for val in actual)
```

Comment:
can we leave the comparison as `val != "new"` since `"new" not in val` may match unnecessary things

Reply:
Is it possible that an erroneous `"new"` was not caught because newlines were not preserved correctly? Now that they should be, the value might not be *exactly equal* to `"new"`?

Reply:
I changed it because of the comment right above.  `val` is like `NEW_VAR=` so even if `new` happen to be the value of `val`, it would look like `NEW_VAR=new` so the original comparison predicate `val != "new"` will always be true meaning the test will never catch any bugs.

This fix has nothing to do with all my code changes. I simply catch the bug that the test is invalid.

Reply:
Oh I see. Ok thanks for the catch. 

Please update it to:

```python
assert all(val.split("=")[-1] != "new" for val in actual)
```

so that the intent of the test is kept (this is testing that the value of the variable is not set to `new`, not that `new` doesn't occur in either the value or the variable name).

Reply:
done in 6b280b88811c34171e57c161d7f03d212fc1a6f8